packages/next/lib/eslint/hasNecessaryDependencies.ts

```diff
@@ -0,0 +1,68 @@
+import chalk from 'chalk'
+import path from 'path'
+
+import { ESLintFatalError } from './ESLintFatalError'
+import { getOxfordCommaList } from '../oxford-comma-list'
+
+const requiredESLintPackages = [
+  'eslint',
+  '@babel/core',
+  '@babel/eslint-parser',
+  'eslint-config-next',
+  '@next/eslint-plugin-next',
+  'eslint-plugin-react',
+  'eslint-plugin-react-hooks',
+  'eslint-plugin-import',
+]
+
+const requiredTSPackages = ['@typescript-eslint/parser']
+
+export type NecessaryDependencies = {
+  resolvedESLint: string
+}
+
+export async function hasNecessaryDependencies(
+  baseDir: string,
+  eslintrcFile: string | null,
+  isUsingTS: boolean
+): Promise<NecessaryDependencies> {
+  let resolutions = new Map<string, string>()
+
+  // If TypeScript is being used, ensure that @typescript-eslint/parser is also installed
+  const requiredPackages = isUsingTS
+    ? requiredESLintPackages.concat(requiredTSPackages)
+    : requiredESLintPackages
+
+  const missingPackages = requiredPackages.filter((pkg) => {
+    try {
+      resolutions.set(pkg, require.resolve(pkg, { paths: [baseDir] }))
+      return false
+    } catch (_) {
+      return true
+    }
+  })
+
+  if (missingPackages.length < 1) {
+    return { resolvedESLint: resolutions.get('eslint')! }
+  }
+
+  const removalLocation = eslintrcFile
+    ? chalk.cyan(path.basename(eslintrcFile)) + ' file from your application'
+    : chalk.cyan('eslintConfig') + ' field from your package.json file'
+
+  throw new ESLintFatalError(
+    chalk.bold.red(
+      `It looks like you're trying to use ESLint but do not have the following required package(s) installed:`
+    ) +
+      `\n\n\t` +
+      getOxfordCommaList(missingPackages) +
+      '\n\n' +
+      chalk.bold(`Please install all required dependencies by running:`) +
+      '\n\n' +
+      `\t${chalk.bold.cyan('npx install-peerdeps --dev eslint-config-next')}` +
```

Comment:
Instead of relying on an external package to show this, maybe we can tweak/re-use the lib we have for checking for required dependencies for `typescript` https://github.com/vercel/next.js/blob/25ca1b5babe41033f4a2c9e5bbc98c384c8f8660/packages/next/lib/typescript/hasNecessaryDependencies.ts

Reply:
Done. Tweaked the file, moved it file a level up and used it instead